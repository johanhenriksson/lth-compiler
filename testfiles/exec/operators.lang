int main() {
    int a = (22 - 3) * 2 / 2 % 3;
    float b = (22.0 - 3.5) * 2.1 / 2.3 % 3.75;
    return 0;
}
