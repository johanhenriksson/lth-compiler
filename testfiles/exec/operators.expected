; Library Functions
declare void @print(i64)
declare i64 @read()
declare i64 @truncate(double)
declare double @promote(i64)
declare void @assert(i1)
declare void @except(i1)

define i64 @main() {
%a = alloca i64
%_t4 = sub i64 22, 3
%_t3 = mul i64 %_t4, 2
%_t2 = sdiv i64 %_t3, 2
%_t1 = srem i64 %_t2, 3
store i64 %_t1, i64* %a
%b = alloca double
%_t8 = fsub double 22.0, 3.5
%_t7 = fmul double %_t8, 2.1
%_t6 = fdiv double %_t7, 2.3
%_t5 = fdiv double %_t6, 3.75
store double %_t5, double* %b
ret i64 0
}

